### ISSUE TITLE
Megatron Supports InternVL OR NOT?

### ISSUE THREAD

**Issue by robotsp** [NVIDIA/Megatron-LM#1921]
Do we have supporting ways on using Megatron-LM to train InternVL?
Including, 
1. internvl coverter tool from hf->megatron
2. runable model training scripts

**Comment by sbhavani** [comment]
@robotsp [Megatron Bridge](https://github.com/NVIDIA-NeMo/Megatron-Bridge) is where we have Hugging Face checkpoint conversion and new model support. If you have any model requests please feel free to create an issue/PR there.

For InternVL3, we have an implementation here: https://github.com/ISEEKYAN/mbridge/tree/main/example/internvl3. Note that mbridge models are being consolidated into Megatron Bridge.

**Comment by robotsp** [comment]
> [@robotsp](https://github.com/robotsp) [Megatron Bridge](https://github.com/NVIDIA-NeMo/Megatron-Bridge) is where we have Hugging Face checkpoint conversion and new model support. If you have any model requests please feel free to create an issue/PR there.
> 
> For InternVL3, we have an implementation here: https://github.com/ISEEKYAN/mbridge/tree/main/example/internvl3. Note that mbridge models are being consolidated into Megatron Bridge.

Thanks :) May I know how can I integrate the mbridge model into Megatron-LM for the training process, do you have any guidelines? @sbhavani

**Comment by sbhavani** [comment]
@robotsp  we are still working on a guide but you can see an example here of how to load a HF model into Megatron: https://github.com/NVIDIA-NeMo/Megatron-Bridge/tree/main/src/megatron/bridge/models#loading-a-hugging-face-model-into-megatron

**Comment by robotsp** [comment]
> [@robotsp](https://github.com/robotsp) we are still working on a guide but you can see an example here of how to load a HF model into Megatron: https://github.com/NVIDIA-NeMo/Megatron-Bridge/tree/main/src/megatron/bridge/models#loading-a-hugging-face-model-into-megatron

Great, I think we should create a pretrain_xxx.py in megatron and define model_provider, forward func, loss func and etc for the training, where model provider we should use the mbridge one, right?

**Comment by sbhavani** [comment]
@robotsp  were you able to get InternVL working with Megatron-LM? We are working on a general example for using Megatron Bridge with Megatron-LM as you described.

**Comment by robotsp** [comment]
Yes, I am implementing the training process of InternVL3 with Megatron-LM. I am appreciated to join the work on a general example as well. Pls let me know if I could help :) @sbhavani

**Comment by sbhavani** [comment]
Great! Feel free to create a PR in [Megatron Bridge](https://github.com/NVIDIA-NeMo/Megatron-Bridge). We would love to have an example there

**Comment by robotsp** [comment]
> Great! Feel free to create a PR in [Megatron Bridge](https://github.com/NVIDIA-NeMo/Megatron-Bridge). We would love to have an example there

Cool, btw does mbridge support HF->MM weight transform while pipeline parallelism > 1？I cannot find the implementation in InternVL3. @sbhavani

**Comment by sbhavani** [comment]
@robotsp Megatron Bridge supports PP>1 but this happens after the weight transformation. We plan to add InternVL3 (https://github.com/NVIDIA-NeMo/Megatron-Bridge/issues/1297) with end to end recipes for pre-training. Please add any requirements you have in Megatron Bridge.
